volatility3/framework/plugins/windows/psscan.py

```diff
@@ -269,39 +269,60 @@ def _generator(self):
             filter_func=pslist.PsList.create_pid_filter(self.config.get("pid", None)),
         ):
             file_output = "Disabled"
-            if self.config["dump"]:
-                # windows 10 objects (maybe others in the future) are already in virtual memory
-                if proc.vol.layer_name == kernel.layer_name:
-                    vproc = proc
+
+            # windows 10 objects (maybe others in the future) are already in virtual memory
+            # if the proc native_layer_name and layer_name match then it is in 'virtual' memory.
+            if proc.vol.layer_name == proc.vol.native_layer_name:
```

Comment:
This isn't a robust test (it only works when `scan_processes` returns carved processes), and I don't really want people using it to determine whether an address is physical or not.  The poolscanner's been asked to scan the kernel layer, so... it should only ever return virtual results?  Have you verified that the "else" branch ever gets hit?